src/odb/src/lefin/lefTechLayerPitchRuleParser.cpp

```diff
@@ -0,0 +1,108 @@
+ * (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES;
+ * LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND
+ * ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
+ * (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF THIS
+ * SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ */
+
+#include <functional>
+#include <iostream>
+#include <string>
+
+#include "boostParser.h"
+#include "db.h"
+#include "lefLayerPropParser.h"
+#include "lefin.h"
+
+namespace odb {
+
+lefTechLayerPitchRuleParser::lefTechLayerPitchRuleParser(lefin* l)
+{
+  lefin_ = l;
+}
+
+void lefTechLayerPitchRuleParser::parse(std::string s, odb::dbTechLayer* layer)
+{
+  odb::dbTechLayerPitchRule* rule = odb::dbTechLayerPitchRule::create(layer);
+
+  qi::rule<std::string::iterator, space_type> PITCH
+      = ((lit("PITCH") >> double_
+          >> double_)[boost::bind(&odb::lefTechLayerPitchRuleParser::setPitchXY,
+                                  this,
+                                  _1,
+                                  rule,
+                                  &odb::dbTechLayerPitchRule::setPitchX,
+                                  &odb::dbTechLayerPitchRule::setPitchY)]
+         | lit("PITCH") >> double_[boost::bind(
+               &odb::lefTechLayerPitchRuleParser::setPitch,
+               this,
+               _1,
+               rule,
+               &odb::dbTechLayerPitchRule::setPitch)]);
+
+  qi::rule<std::string::iterator, space_type> FIRST_LAST_PTICH
+      = (PITCH >> (lit("FIRSTLASTPITCH") >> double_[boost::bind(
+                       &odb::lefTechLayerPitchRuleParser::setFirstLastPitch,
+                       this,
+                       _1,
+                       rule,
+                       &odb::dbTechLayerPitchRule::setFirstLastPitch)])
+         >> lit(";"));
+  auto first = s.begin();
+  auto last = s.end();
+  bool valid
+      = qi::phrase_parse(first, last, FIRST_LAST_PTICH, space) && first == last;
+  if (!valid) {
+    lefin_->warning(281,
+                    "parse mismatch in layer propery LEF58_Pitch for layer");
+    odb::dbTechLayerPitchRule::destroy(rule);
+  }
+}
+
+void lefTechLayerPitchRuleParser::setFirstLastPitch(
+    double val,
+    odb::dbTechLayerPitchRule* rule,
+    void (odb::dbTechLayerPitchRule::*func)(int))
+{
+  (rule->*func)(lefin_->dbdist(val));
+}
+
+void lefTechLayerPitchRuleParser::setPitch(
+    double val,
+    odb::dbTechLayerPitchRule* rule,
+    void (odb::dbTechLayerPitchRule::*func)(int))
+{
+  (rule->*func)(lefin_->dbdist(val));
+}
+
+void lefTechLayerPitchRuleParser::setPitchXY(
+    boost::fusion::vector<double, double>& params,
+    odb::dbTechLayerPitchRule* rule,
+    void (odb::dbTechLayerPitchRule::*func1)(int),
+    void (odb::dbTechLayerPitchRule::*func2)(int))
+{
+  (rule->*func1)(lefin_->dbdist(at_c<0>(params)));
+  (rule->*func2)(lefin_->dbdist(at_c<1>(params)));
+}
+
```

Comment:
I think you are copying the implementation of function binding but out of context here. The idea of having a class method with the rule's method as an argument is to avoid having multiple parser methods that does the same thing but only changes the called rule's method.
You should do one of two things:
- Have only one method "setInt" that takes the rule's method as an argument
- Do not pass the rule method as an argument and call it directly in the parser method.